packages/odyssey-react-mui/src/OdysseyProvider.tsx

```diff
@@ -51,32 +72,66 @@ const OdysseyProvider = <SupportedLanguages extends string>({
   stylisPlugins,
   themeOverride,
   translationOverrides,
-}: OdysseyProviderProps<SupportedLanguages>) => (
-  <OdysseyCacheProvider
-    emotionRoot={emotionRoot}
-    emotionRootElement={emotionRootElement}
-    hasShadowDom={Boolean(shadowRootElement || shadowDomElement)}
-    nonce={nonce}
-    stylisPlugins={stylisPlugins}
-  >
-    <OdysseyThemeProvider
-      designTokensOverride={designTokensOverride}
-      shadowRootElement={shadowRootElement || shadowDomElement}
-      themeOverride={themeOverride}
+}: OdysseyProviderProps<SupportedLanguages>) => {
+  const { isLowContrast } = useBackground();
+
+  const odysseyTokens = useMemo(
+    () => ({ ...Tokens, ...designTokensOverride }),
+    [designTokensOverride],
+  );
+
```

Comment:
Instead of having `OdysseyProvider` consume `BackgroundProvider`, it should be baked in.

Also, we need a hook to calculate the background color of the parent element of `OdysseyProvider`. It needs to output a `div` for that to work.

We'd get the `ref` of the div and check against the DOM to compute the `background-color` of it.

Once we have that, the rest can be changed by having `Surface` and other "high contrast" components consume the `BackgroundProvider` and change the value.